compiler/bcq-tools/generate_bcq_output_arrays.py

```diff
@@ -255,6 +255,196 @@ def print_bcq_output_arrays(flags):
+                if nodename == op.inputs[i].name:
+                    if op.type in mark_type:
+                        if op.outputs[0].name[:-2] not in fuseop_prefix_dict.keys():
+                            fuseop_prefix_dict[op.outputs[0].name[:-2]] = set()
+                        fuseop_prefix_dict[op.outputs[0].name[:-2]].add(prefix)
+                        if op.outputs[0].name not in visited_nodes:
+                            queue_prefix.append(prefix)
+                            queue_nodename.append(op.outputs[0].name)
+                            visited_nodes.add(op.outputs[0].name)
+                    elif op.type in mark_and_stop_type:
+                        if op.outputs[0].name[:-2] not in fuseop_prefix_dict.keys():
+                            fuseop_prefix_dict[op.outputs[0].name[:-2]] = set()
+                        fuseop_prefix_dict[op.outputs[0].name[:-2]].add(prefix)
+                    elif op.type in pass_type and op.outputs[0].name not in visited_nodes:
+                        queue_prefix.append(prefix)
+                        queue_nodename.append(op.outputs[0].name)
+                        visited_nodes.add(op.outputs[0].name)
+
+    # the name of metadata node
+    ret_output_arrays = ['one_compiler/bcqinfo_one_metadata']
+
+    # given node from user
+    ret_output_arrays.append(output_arrays)
+
+    # all pairs of candidate operations and related BCQ information nodes
+    for prefix in prefix_set:
+        for fusable_op in prefix_node_dict[prefix]:
+            ret_output_arrays.append(prefix + '/bcqinfo_do_w_x')
+            ret_output_arrays.append(prefix + '/bcqinfo_alpha')
+            ret_output_arrays.append(prefix + '/bcqinfo_packed_binary_code')
+            ret_output_arrays.append(prefix + '/bcqinfo_number_of_clusters')
+            ret_output_arrays.append(prefix + '/bcqinfo_size_of_clusters')
+            ret_output_arrays.append(prefix + '/bcqinfo_qbits_of_clusters')
+            ret_output_arrays.append(fusable_op)
+            if has_dequant_weight:
+                ret_output_arrays.append(prefix + '/bcqinfo_dequant_weight')
+    for fuseop in fuseop_prefix_dict.keys():
+        if len(fuseop_prefix_dict[fuseop]) == 1:
+            prefix = fuseop_prefix_dict[fuseop].pop()
+            ret_output_arrays.append(prefix + '/bcqinfo_do_w_x')
+            ret_output_arrays.append(prefix + '/bcqinfo_alpha')
+            ret_output_arrays.append(prefix + '/bcqinfo_packed_binary_code')
+            ret_output_arrays.append(prefix + '/bcqinfo_number_of_clusters')
+            ret_output_arrays.append(prefix + '/bcqinfo_size_of_clusters')
+            ret_output_arrays.append(prefix + '/bcqinfo_qbits_of_clusters')
+            ret_output_arrays.append(fuseop)
+            if has_dequant_weight:
+                ret_output_arrays.append(prefix + '/bcqinfo_dequant_weight')
+
+    return ret_output_arrays
+
+
+def get_bcq_output_arrays(input_path, output_arrays):
```

Comment:
> Why is this added?

Maybe I should have asked here...  this file is in `bcq-tools` not in `one-cmds`

Reply:
Oh, right. I will split this file when I post a PR.